axlearn/common/compiler_options.py

```diff
@@ -92,16 +70,41 @@ def default_xla_options(
             # though, may make the scheduler weighting too much on the memory usage instead
             # of latency side.
             xla_latency_hiding_scheduler_rerun=2,
-            xla_tpu_use_enhanced_launch_barrier=True,
+            # Improved performance for v6e.
+            xla_tpu_scoped_vmem_limit_kib=98304,
+        )
+        options.update(
```

Comment:
need to group these fileds by value type to pass pytype check otherwise, pytype fails with something like

```
File "/Users/ethanli/Workspace/foundation-models/axlearn/axlearn/common/compiler_options.py", line 48, in default_xla_options: Function dict.update was called with the wrong arguments [wrong-arg-types]
         Expected: (self, megascale_grpc_premap_memory_bytes, xla_enable_async_all_gather: bool, ...)
  Actually passed: (self, megascale_grpc_premap_memory_bytes, xla_enable_async_all_gather: str, ...)

```